dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java

```diff
@@ -238,6 +242,24 @@ protected void updateDBWorkflowWithSourceControlWorkflow(Workflow workflow, Work
         }
     }
 
+    /**
+     * Handle webhooks from GitHub apps after branch deletion (redirected from AWS Lambda)
+     * - Delete version for corresponding service and workflow
+     * @param repository Repository path (ex. dockstore/dockstore-ui2)
+     * @param gitReference Git reference from GitHub (ex. refs/tags/1.0)
+     * @return List of updated workflows
+     */
+    protected List<Workflow> githubWebhookDelete(String repository, String gitReference) {
+        // Retrieve name from gitReference
+        Pair<String, String> referencePair = parseGitHubReference(gitReference);
+
+        // Find all workflows and services that are github apps and use the given repo
+        List<Workflow> workflows = workflowDAO.findAllByPath("github.com/" + repository, false).stream().filter(workflow -> Objects.equals(workflow.getMode(), DOCKSTORE_YML)).collect(
```

Comment:
don't get everything and then filter by mode, it's costly

Reply:
Is there a findAllByPath function that has mode as a param?

Reply:
no